src/main/common/vector.h

```diff
@@ -43,34 +43,46 @@ typedef union vector3_u {
     };
 } vector3_t;
 
-typedef struct matrix33_s {
+typedef union matrix33_u {
     float m[3][3];
+    struct {
+        vector3_t x;
+        vector3_t y;
+        vector3_t z;
+    };
+    struct {
```

Comment:
this alias hides internal structure (row/column ordering). In some places it may decrease readability (any it will look strange to someone working with rotation/transformation matrices in other programs)